src/screens/ConstructionSiteDetailScreen.js

```diff
@@ -0,0 +1,162 @@
+import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import { ScrollView, StyleSheet, TouchableOpacity, View } from 'react-native';
+
+import { colors, consts, normalize } from '../config';
+import {
+  BoldText,
+  Icon,
+  Image,
+  RegularText,
+  SafeAreaViewFlex,
+  Title,
+  TitleContainer,
+  WebViewMap,
+  Wrapper,
+  WrapperWithOrientation
+} from '../components';
+import { arrowLeft, locationIconAnchor, location as locationIcon } from '../icons';
+import { useMatomoTrackScreenView } from '../hooks';
+import { ConstructionSiteContext } from '../ConstructionSiteProvider';
+import { momentFormat } from '../helpers';
+import moment from 'moment';
+
+const { MATOMO_TRACKING } = consts;
+
+const formatDates = (startDate, endDate) => {
+  const formattedStartDate = momentFormat(startDate, 'DD.MM.YYYY');
+  const startTime = moment(startDate, 'YYYY-MM-DD HH:mm:ss').format('HH:mm');
+
+  const formattedEndDate = endDate ? momentFormat(endDate, 'DD.MM.YYYY') : undefined;
+  const endTime = endDate ? moment(endDate, 'YYYY-MM-DD HH:mm:ss').format('HH:mm') : undefined;
+
+  if ((endTime !== undefined && endTime !== '00:00') || startTime !== '00:00') {
+    return formattedEndDate && endTime
+      ? `${formattedStartDate} ${startTime} - ${formattedEndDate} ${endTime}`
+      : `${formattedStartDate} ${startTime}`;
+  }
+
+  return formattedEndDate ? `${formattedStartDate} - ${formattedEndDate}` : formattedStartDate;
+};
+
+export const ConstructionSiteDetailScreen = ({ navigation }) => {
+  const { constructionSites } = useContext(ConstructionSiteContext);
+
+  const index = navigation.getParam('index');
```

Comment:
this ensures, that the screen does not break if there is a navigation without index param? can this happen somehow? how can i test it?

Reply:
you can add a dummy element somewhere that lets you navigate to the screen with wrong props.

currently it should not be able to go wrong, because we have only one place where we navigate to the screen.

It could happen due to more navigations being added later on, or in the process of refactoring.

Reply:
do we want that for other routes as well?